### PR TITLE
feat: add create organization overlay

### LINKED ISSUE #5899: Multi Org Phase 2 - add new modal and functionality to create organization
Mock: https://www.figma.com/file/fvVSKwnepmRWvzFlIxCWCe/Multi-org?node-id=1967%3A20534

Add new modal according to the figma mockups to create an organization.

Details:
* Perform duplicate org name check using organizations list in identity.
* Update openapi sha to use the Post Organization route.

Pending Questions for @juliajames12 :
- If there's a conflict in the name of the organization when creating one, do we still display a Toast with Retry or should we display an inline error under the org name textbox?
- Do we allow `Cancel` while creating an organization?
- Do we straight-away go to the new organization page or should we just display the Toast with `Switch to Organization` link as displayed in the mockup?

### PR DESCRIPTION
**Pull request by wdoconnell**
Closes #5899.

Adds a modal to the UI to permit user to create orgs (if allowed).

### Checklist

Authors and Reviewer(s), please verify the following:

- [X] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [X] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [X] Documentation updated or issue created (provide link to issue/PR)
- [X] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [X] Feature flagged, if applicable

### CODE REVIEW

File: src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/MenuItem.tsx [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 
 export const CreateOrganizationMenuItem: FC = () => {
+  const dispatch = useDispatch()
   const handleCreateOrg = () => {
-    // TODO: Enable this when #5899 is worked on
-    // event('globalheader_createOrg_menuItem_clicked')
-    // TODO: Remove this comment when you are working on https://github.com/influxdata/ui/issues/5899
-    //       Add create organization functionality with modal, error handling and redirect to new organization
+    dispatch(
+      showOverlay('create-organization', null, () => dispatch(dismissOverlay()))
```
Comment (blegesse-git) [inline src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/MenuItem.tsx:19]: I don't think you need the second `dispatch()` call. 
```suggestion
      dispatch(showOverlay('create-organization', null, () => (dismissOverlay()))
```

File: src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/RegionDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+// Libraries
+import React, {Dispatch, FC} from 'react'
+import {Dropdown} from '@influxdata/clockface'
+
+// Types
+import {Cluster} from 'src/client/unityRoutes'
+
+interface Props {
+  regions: Cluster[]
+  currentRegion: string
```
Comment (blegesse-git) [inline src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/RegionDropdown.tsx:10]: minor - alphabetize?

File: src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/RegionDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  setCurrentRegion: Dispatch<any>
```
Comment (blegesse-git) [inline src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/RegionDropdown.tsx:11]:
```suggestion
  setCurrentRegion: (regionID: string) => void
```

File: src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationOverlay.tsx [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+      setValidationMsg(OrgOverlayValidationError.NameConflictError)
+      setCreateOrgButtonStatus(ComponentStatus.Disabled)
+    } else {
+      setValidationMsg(OrgOverlayValidationError.None)
+      setCreateOrgButtonStatus(ComponentStatus.Default)
+    }
+  }
+
+  // Ajax requests
+  useEffect(() => {
```
Comment (blegesse-git) [inline src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationOverlay.tsx:180]: this is more of a personal preference than anything but I'd move these useEffect hooks to the top of the component where the states are being set

**Review by blegesse-git — APPROVED** [review]
🥳 Looks good to me. Comments are mostly suggestions and/or minor adjustments that can be made.

File: src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrgInput.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+export const CreateOrgInput: FC<Props> = ({
+  handleValidateOrgName,
+  orgName,
+  validationMsg,
+}) => {
+  const [userTouchedFormInput, setUserTouchedFormInput] = useState(false)
+
+  const handleInputChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
```
[the remark this answers is not in the document]
Comment (wdoconnell) [reply src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrgInput.tsx:30]: I hear you, and good idea, but I would use `evt` over `event` because `event` is a specialized term in the codebase for the eventing function that sends events to snowflake. If we rename regular handler events to `event`, that'll make those hard to distinguish when someone is running a Control+F searching just for the `event` calls that handle reporting.

File: src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationOverlay.tsx [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+        region: currentRegion,
+      })
+      setCreateOrgButtonStatus(ComponentStatus.Default)
+      onClose()
+      dispatch(notify(orgCreateSuccess()))
+      // Will make another API call to the allowance endpoint in PR resolving this issue.
+      // https://github.com/influxdata/ui/issues/6267
+      // Need a parallel update to global header state to ensure list of orgs remains updated.
+      dispatch(getQuartzOrganizationsThunk(currentAccountId))
+    } catch (err) {
```
[the remark this answers is not in the document]
Comment (wdoconnell) [reply src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationOverlay.tsx:139]: I hear you, but this falls into the same category as `evt` versus `event` in my view. I'd like to avoid calling it `error` because a Control+F (esp. if not case sensitive) will pull up actual references to the `Error` constructor as opposed to errors in try/catch blocks.

File: src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationOverlay.tsx [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+
+  const handleSelectProvider = (
+    providerID: OrganizationCreateRequest['provider']
+  ) => {
+    setCurrentProvider(providerID)
+    const defaultRegionId = providerMap[providerID][0].regionId
+    setCurrentRegion(defaultRegionId)
+  }
+
+  const handleValidateOrgName = (proposedOrgName: string) => {
```
[the remark this answers is not in the document]
Comment (wdoconnell) [reply src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationOverlay.tsx:161]: Great point - that hadn't occurred to me. Let me check with the team on this.

File: src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/MenuItem.tsx (hunk 1 above)
Comment (wdoconnell) [reply src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/MenuItem.tsx:19]: That's true, but not for the reason we'd expect. 

The third argument to `showOverlay` is a function that is run `onClose`. However, the context provider always provides an `onClose` function that dispatches `dismissOverlay()` regardless of what we provide as the third value. I've tested it, for example, with `null` as the third argument, and there's zero change in behavior - the overlay still dismisses as expected.

TLDR, this is a bigger problem with the `showOverlay` and overlay context provider because the third argument isn't providing value. I'd like to leave `dispatch(dismissOverlay())))` in there because it at least provides some descriptive account of what is going on.

File: src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/RegionDropdown.tsx (hunk 2 above)
Comment (wdoconnell) [reply src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/RegionDropdown.tsx:10]: Good catch

**Review by hansenreid — APPROVED** [review]
